web/satellite/src/components/account/billing/paymentMethods/AddTokenCard.vue

```diff
@@ -2,12 +2,227 @@
+                    class="token__add-funds__button"
+                    label="Continue to CoinPayments"
+                    width="150px"
+                    height="30px"
+                    font-size="11px"
+                    :on-press="onConfirmAddSTORJ"
+                />
+                <VButton
+                    v-if="totalCount > 0"
+                    class="token__add-funds__button"
+                    label="Back"
+                    is-transparent="true"
+                    width="50px"
+                    height="30px"
+                    font-size="11px"
+                    :on-press="toggleShowAddFunds"
+                />
+            </div>
+        </div>
+    </div>
 </template>
 
 <script lang="ts">
-import { Vue } from 'vue-property-decorator';
+import { Component, Prop, Vue } from 'vue-property-decorator';
 
+import StorjLarge from '@/../static/images/billing/storj-icon-large.svg';
+import VButton from '@/components/common/VButton.vue';
+import TokenDepositSelection from '@/components/account/billing/paymentMethods/TokenDepositSelection.vue';
+
+import { PAYMENTS_ACTIONS } from '@/store/modules/payments';
+import { PaymentAmountOption } from '@/types/payments';
+
+const {
+    MAKE_TOKEN_DEPOSIT,
+    GET_PAYMENTS_HISTORY,
+} = PAYMENTS_ACTIONS;
+
+// @vue/component
+@Component({
+    components: {
+        StorjLarge,
+        VButton,
+        TokenDepositSelection,
+    },
+})
 export default class AddTokenCard extends Vue {
+    @Prop({default: 0})
+    private readonly totalCount: number;
+    @Prop({default: false})
+    private readonly DEFAULT_TOKEN_DEPOSIT_VALUE = 10; // in dollars.
+    private readonly MAX_TOKEN_AMOUNT = 1000000; // in dollars.
+    private tokenDepositValue: number = this.DEFAULT_TOKEN_DEPOSIT_VALUE;
+
+    public toggleShowAddFunds(): void {
+        this.$emit("showAddFunds")
+    }
+
+    /**
+     * Set of default payment options.
+     */
+    public paymentOptions: PaymentAmountOption[] = [
+        new PaymentAmountOption(10, `USD $10`),
+        new PaymentAmountOption(20, `USD $20`),
+        new PaymentAmountOption(50, `USD $50`),
+        new PaymentAmountOption(100, `USD $100`),
+        new PaymentAmountOption(1000, `USD $1000`),
+    ];
+
+    /**
+     * onConfirmAddSTORJ checks if amount is valid.
+     * If so processes token payment and returns state to default.
+     */
+    public async onConfirmAddSTORJ(): Promise<void> {
+
+        if (!this.isDepositValueValid) return;
+
+        try {
+            const tokenResponse = await this.$store.dispatch(MAKE_TOKEN_DEPOSIT, this.tokenDepositValue * 100);
+            await this.$notify.success(`Successfully created new deposit transaction! \nAddress:${tokenResponse.address} \nAmount:${tokenResponse.amount}`);
+            const depositWindow = window.open(tokenResponse.link, '_blank');
+            if (depositWindow) {
+                depositWindow.focus();
+            }
+            location.reload()
```

Comment:
why do we need that?

Reply:
we could just refetch data if needed instead of reloading the whole app

Reply:
Re-worked to not use reload.